### ISSUE TITLE
Improve Proxy Protocol docs/configuration parsing

### ISSUE THREAD

**Issue by sidewinder12s** [hashicorp/vault#4065]
**Feature Request:**
Improve the documentation/parsing of proxy_protocol_authorized_addrs listener configuration options. The parsing is not very friendly and the documentation just says the field is a string, leaving you guessing how to input multiple IPs which is completely expected if your putting in IPs of a load balancer. I would have expected it to be an array. The fact you need to separate the addresses with a comma isn't documented either.

**Environment:**

* Vault Version: v0.9.3
* Operating System/Architecture: docker/ubuntu/amd64

**Vault Config File:**
Within the listener config:
```
proxy_protocol_behavior = "allow_authorized"
proxy_protocol_authorized_addrs = ""10.15.163.13:8200, 10.15.163.12:8200""
```

**Startup Log Output:**
```
{\"line\":\"\u001b[0;31mError initializing listener of type tcp: failed parsing proxy_protocol_authorized_addrs: error parsing authorized address: Unable to convert \\"  10.15.163.12:8200\\" to an IPv4 or IPv6 address, or a UNIX Socket\u001b[0m\",\"source\":\"stdout\",\"tag\":\"vault:0.9.3/container_name/9882051d8d99\"},\"time\":\"1520029773.110934\",\"host\":\"host.com\"}
```
**Expected Behavior:**
Vault should have started up.

**Actual Behavior:**
Vault failed to start

**Steps to Reproduce:**
Add a proxy_protocol_authorized_addrs field with 2 addresses and a space between the 2 to the listener config.
```
proxy_protocol_authorized_addrs = ""10.15.163.13:8200, 10.15.163.12:8200""
```

**References:**
Removing the space between the 2 addresses fixes this however that doesn't seem to be documented anywhere.
```
proxy_protocol_authorized_addrs = ""10.15.163.13:8200,10.15.163.12:8200""
```

**Comment by shelby-moore-hs** [comment]
It's also strange that with `proxy_protocol_behavior = "use_always"` you need to specify `proxy_protocol_authorized_addrs`, which will cause Vault to not start up if it is an empty string. It would be preferable to not require `proxy_protocol_authorized_addrs` at all in this case, or at least allow it to be an empty string.

**Comment by jefferai** [comment]
Closed by #4528
